widget/progressbarinfinite.go

```diff
@@ -111,49 +107,62 @@ func (p *infProgressRenderer) stop() {
 	p.progress.propertyLock.Lock()
 	defer p.progress.propertyLock.Unlock()
 
-	p.running = false
+	p.wasRunning = false
 	p.animation.Stop()
 }
 
 func (p *infProgressRenderer) Destroy() {
+	p.progress.propertyLock.Lock()
+	p.progress.running = false
+	p.progress.propertyLock.Unlock()
+
 	p.stop()
 }
 
 // ProgressBarInfinite widget creates a horizontal panel that indicates waiting indefinitely
 // An infinite progress bar loops 0% -> 100% repeatedly until Stop() is called
 type ProgressBarInfinite struct {
 	BaseWidget
+	running bool
 }
 
 // Show this widget, if it was previously hidden
 func (p *ProgressBarInfinite) Show() {
-	p.Start()
+	p.propertyLock.Lock()
+	p.running = true
+	p.propertyLock.Unlock()
+
 	p.BaseWidget.Show()
 }
 
 // Hide this widget, if it was previously visible
 func (p *ProgressBarInfinite) Hide() {
-	p.Stop()
+	p.propertyLock.Lock()
+	p.running = false
+	p.propertyLock.Unlock()
+
 	p.BaseWidget.Hide()
 }
 
 // Start the infinite progress bar animation
 func (p *ProgressBarInfinite) Start() {
-	cache.Renderer(p).(*infProgressRenderer).start()
+	p.SetFieldsAndRefresh(func() {
```

Comment:
as discussed in Slack, let's just use the basewidget internal lock directly, instead of SetFieldsAndRefresh, in case we decide to remove or deprecate it

Reply:
Yup, good point. That did also made make me realise that we don't need to refresh if we are already running or already stopped :)

Reply:
That was easier to implement without using SetFieldsAndRefresh.